server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -443,7 +444,8 @@ public RemoteClusterStateManifestInfo writeIncrementalMetadata(
             uploadedMetadataResults,
             previousManifest.getPreviousClusterUUID(),
             clusterStateDiffManifest,
-            false
+            false,
+            previousManifest.getCodecVersion()
```

Comment:
What happens if between the previous manifest and current upload, a node with lower version gets added to the cluster?

Reply:
if a node is added to the cluster, cluster state will get updated and manifest will also change accordingly

Reply:
But we will still use codec from previous manifest, can that lead to incorrect version being picked?

Reply:
If a node with lower version gets added and the codec for that version is also lower than current manifest, then will write full cluster state. Incremental cluster state will be written when codec version in previous manifest and that for current cluster state nodes is the same.